backend/pkg/api/connect/service/transform/service.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package transform
+
+import (
+	"connectrpc.com/connect"
+	"context"
+	"errors"
+	apierrors "github.com/redpanda-data/console/backend/pkg/api/connect/errors"
+	"github.com/redpanda-data/console/backend/pkg/config"
+	v1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1"
+	"github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1/dataplanev1alpha1connect"
+	"github.com/redpanda-data/console/backend/pkg/redpanda"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+)
+
+var _ dataplanev1alpha1connect.TransformServiceHandler = (*Service)(nil)
+
+type Service struct {
+	cfg         *config.Config
+	redpandaSvc *redpanda.Service
+}
+
+func (s *Service) DeployTransform(ctx context.Context, c *connect.Request[v1alpha1.DeployTransformRequest]) (*connect.Response[v1alpha1.DeployTransformResponse], error) {
+	if !s.cfg.Redpanda.AdminAPI.Enabled {
+		return nil, apierrors.NewConnectError(
+			connect.CodeUnimplemented,
+			errors.New("the redpanda admin api must be configured to deploy a transform"),
+			apierrors.NewErrorInfo(v1alpha1.Reason_REASON_FEATURE_NOT_CONFIGURED.String()),
+			apierrors.NewHelp(apierrors.NewHelpLinkConsoleReferenceConfig()),
+		)
+	}
```

Comment:
super minor nit, but maybe make this into it's own function so we can just do something like:
```go
if err := checkAdminAPIEnabled(errors.New("specific message here")); err != nil {
  return nil, err
}

Reply:
i mirrored the pattern in user. wouldn't mind having a function that does that check in a util package or something somewhere (sorta like the auth hooks) and then sharing it between the two/other locs. because i agree it definitely feels repetitious.

Reply:
or dump it in some middleware somewhere 🤷‍♀️

Reply:
I wouldn't put it in middleware. I think it's fine to leave it as is too. It was just a thought. Not a big deal. Fine the way it is too.